clicache/src/DataInput.hpp

```diff
@@ -691,6 +691,7 @@ namespace Apache
         bool m_isRootObjectPdx;
         Apache::Geode::Client::Cache^ m_cache;
         System::Byte* m_buffer;
+        System::Byte* m_allocatedBuffer;
```

Comment:
My only issue is that your new pointer is really just used as a flag. But still seems clear as to its intend.